content/blog/2024-07-01-arrow-2-0.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Arrow 2.0 release
+category: articles
+tags: [core, articles]
+---
+
+# Arrow 2.0 release
+
+We are happy to announce the next major release of Arrow, version 2.0!
+As previously announced, migrating your projects to this release should be hassle-free
+if your code compiled in 1.2.x without any deprecation warnings
+(except for the breaking change in optics generation discussed below).
+
+This release is built with the new K2 compiler, and this gives us the ability
+to support a wider range of platforms, including WebAssembly. From now on, we shall
+provide artifacts for every platform supported by Kotlin.
+
+Apart from stabilization and general bug fixing, the theme of this release
+is improving the different DSLs provided by Arrow libraries. Our goal is to
+empower developers to write more succinct and readable code.
+
+## Simple accumulation in Raise
+
+TODO: Talk about `accumulating`
+
+Using Arrow Core data types as part of serialized data requires additional integration.
+In 1.2.x we started providing compile-time support for `kotlinx.serialization`. 
+From 2.0 we also provide `ArrowModule` for
+[contextual serialization](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/serializers.md#contextual-serialization). This is needed, among others, when the data is processed
+by Ktor.
+
+## Additions to Fx
+
+Writing coroutine-heavy code may become cumbersome over time, especially if
+one intends to use as much concurrency as possible. Arrow Fx includes a `parZip`
+function, but not everybody enjoys having so many brackets.
+
+```kotlin
+parZip(
+    { downloadFile() },
+    { loadDataFromDatabase() }
+) { file, data -> Result(file, data) }
+```
+
+The new `awaitAll` scope tries to improve the situation by tweaking the
+usual `async` mechanism, ensuring that all `Deferred` values are `await`ed
+once the first one is requested. That means that the previous code behaves
+identically to the following.
```

Comment:
I added new documentation in a DRAFT PR that explains the actual issue. This is an example:

```kotlin
coroutineScope {
     val x = async { awaitCancellation() }
     val y = async { throw CancellationException("Boom!") }
     ...
 }
```

This hangs forever, which means this also hangs forever:
```kotlin
either {
  coroutineScope {
     val x = async { awaitCancellation() }
     val y = async { raise("Boom!") }
     ...
   }
 }
```

And `awaitAll` solves that issue by actually awaiting all parallel jobs in parallel. More elaborate here, https://github.com/arrow-kt/arrow/pull/3443/files#diff-eba615458a6921b379f1477f70993424b278b3de927ece7e998c5bf7dba6f7f0R38